File: src/main/java/prosia/basarnas/web/model/map/Placemark.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package prosia.basarnas.web.model.map;

import java.io.Serializable;
import lombok.Data;
import org.primefaces.model.map.LatLng;

/**
 *
 * @author Irfan Rofie
 */
@Data
public class Placemark implements Serializable {

    public static String PLACEMARK_DEFAULT_ICON = "DEFAULT_PLACEMARK";
    private LatLng posisi;
    private String id;
    private String name;
    private String image;
    private String description;
    private Boolean initialState;
    /**
     * Propertise dari placemark saat pertama kali di load dari file akan di
     * simpan dalam properties ini. berfungsi jka placemark ingin diremove dari
     * file.
     */
    private String initialProperties;

    public Placemark() {
    }

    public Placemark(LatLng posisi, String id, String name, String image,
            String description, Boolean initialState, String initialProperties) {
        this.posisi = posisi;
        this.id = id;
        this.name = name;
        this.image = image;
        this.description = description;
        this.initialState = initialState;
        this.initialProperties = initialProperties;
    }
}

File: src/main/webapp/resources/js/map-web/placemark.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
function handlePointClick(event) {
    if (PF('widSobUtil').getJQ().find(':checked').val() === 'PLACEMARK') {
        document.getElementById('idLatPlacemark').value = event.latLng.lat();
        document.getElementById('idLngPlacemark').value = event.latLng.lng();
        remoteNewPlacemark();
    }
}

function updateMarker(lat, lng) {
    document.getElementById('idLatPlacemark').value = lat;
    document.getElementById('idLngPlacemark').value = lng;
}